driver/sparql/Connection.php

```diff
@@ -89,6 +90,38 @@ public function __construct(Client $easy_rdf_client, array $connection_options)
    * {@inheritdoc}
    */
   public function query(string $query, array $args = [], array $options = []): Result {
+    return $this->doQuery($query, $args, $options);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function constructQuery(string $query, array $args = [], array $options = []): Graph {
+    return $this->doQuery($query, $args, $options);
+  }
+
+  /**
+   * Execute the query against the endpoint.
+   *
+   * @param string $query
+   *   The string query to execute.
+   * @param array $args
+   *   An array of arguments for the query.
+   * @param array $options
+   *   An associative array of options to control how the query is run.
+   *
+   * @return \EasyRdf\Sparql\Result|\EasyRdf\Graph
+   *   The query result.
+   *
+   * @throws \InvalidArgumentException
+   *   If $args value is passed but arguments replacement is not yet
+   *   supported. To be removed in #55.
+   * @throws SparqlQueryException
+   *   Exception during query execution, e.g. timeout.
+   *
+   * @see https://github.com/ec-europa/rdf_entity/issues/55
+   */
+  public function doQuery(string $query, array $args = [], array $options = []) {
```

Comment:
Is there a good reason why this should be public?

Reply:
Nope, will fix this in a moment!